### PR TITLE
Add sample application

### PR DESCRIPTION
**Pull request by takecare**
Adds a sample app showcasing the usage of the plugin and some of its features. The inclusion of `detekt` might be particularly interesting since its setup is not straightforward.

Should cover #95

### CODE REVIEW

File: sample/team-props/checkstyle-suppressions.xml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0"?>
+
+<!DOCTYPE suppressions PUBLIC
+  "-//Puppy Crawl//DTD Suppressions 1.1//EN"
+  "http://www.puppycrawl.com/dtds/suppressions_1_1.dtd">
+
+<suppressions>
+
+  <suppress checks=".*" files=".*\.kt" />
```
Comment (mr-archano) [inline sample/team-props/checkstyle-suppressions.xml]: wouldn't be enough to have this setting in the buildscript using the `checkstyle.exclude()`?

Comment (takecare) [reply sample/team-props/checkstyle-suppressions.xml]: indeed. and it's another plugin functionality being showcased.

File: sample/app/build.gradle [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+apply plugin: 'kotlin-android'
+apply plugin: 'kotlin-android-extensions'
+apply plugin: com.novoda.staticanalysis.StaticAnalysisPlugin
+
+android {
+    compileSdkVersion 27
+    buildToolsVersion '27.0.3'
+
+    defaultConfig {
+        applicationId "com.novoda.staticanalysis.sample"
```
Comment (rock3r) [inline sample/app/build.gradle]: Nit: single quotes here and for the `versionName`

File: sample/build.gradle [hunk 3]
```diff
@@ -0,0 +1,20 @@
+buildscript {
+    ext.kotlin_version = '1.2.21'
```
Comment (rock3r) [inline sample/build.gradle]: Why not 1.2.30?

**Comment by mr-archano** [comment]
@rock3r, @takecare: happy to ship this and keep working on the multi-module sample?

**Comment by takecare** [comment]
@mr-archano yup, i'll open another pr for the multi-module setup later

**Comment by rock3r** [comment]
Ok let's 🚢 then! Thanks @takecare :)

**Pull request merged**